src/Nethermind/Nethermind.Trie/BatchedTrieVisitor.cs

```diff
@@ -356,13 +366,28 @@ private void BatchedThread()
     private readonly struct Job
     {
         public readonly ValueHash256 Key;
+        public readonly TNodeContext NodeContext;
         public readonly SmallTrieVisitContext Context;
 
-        public Job(ValueHash256 key, SmallTrieVisitContext context)
+        public Job(ValueHash256 key, TNodeContext nodeContext, SmallTrieVisitContext context)
         {
             Key = key;
+            NodeContext = nodeContext;
             Context = context;
         }
     }
+}
+
+public struct EmptyContext : INodeContext<EmptyContext>
+{
+    public EmptyContext Add(byte[] path) => default;
```

Comment:
I believe default should be `this` instead of `default`. Also, IMO the context mutation should be in `ITreeVisitor`.

Reply:
I think it does not matter and I'm happy to `default` to `this` 😅 

Let me take a look at the latter.

Reply:
As the context is decided when passing the visitor, the caller can use an arbitrary one, so they can match these. I think it leaves a bit more of freedom.